File: internal/cli/key.go
package cli

import (
	"fmt"

	"github.com/spf13/cobra"

	"github.com/morrisclay/scraps-cli/internal/api"
	"github.com/morrisclay/scraps-cli/internal/config"
)

func newKeyCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "key",
		Short: "API key management",
	}

	cmd.AddCommand(newKeyResetRequestCmd())
	cmd.AddCommand(newKeyResetConfirmCmd())

	return cmd
}

func newKeyResetRequestCmd() *cobra.Command {
	var host string

	cmd := &cobra.Command{
		Use:     "reset-request <email>",
		Short:   "Request an API key reset email",
		Example: "  scraps key reset-request user@example.com",
		Args: func(cmd *cobra.Command, args []string) error {
			if len(args) < 1 {
				return fmt.Errorf("email address required\n\nUsage: scraps key reset-request <email>\n\nExample: scraps key reset-request user@example.com")
			}
			return nil
		},
		RunE: func(cmd *cobra.Command, args []string) error {
			email := args[0]

			if host == "" {
				host = config.GetHost()
			}

			client := api.NewClient(host, "")
			if err := client.ResetAPIKeyRequest(email); err != nil {
				return err
			}

			success(fmt.Sprintf("Reset email sent to %s", email))
			info("Check your email for the reset link")
			return nil
		},
	}

	cmd.Flags().StringVarP(&host, "host", "H", "", "Server host")
	return cmd
}

func newKeyResetConfirmCmd() *cobra.Command {
	var host string
	var noLogin bool

	cmd := &cobra.Command{
		Use:     "reset-confirm <token>",
		Short:   "Confirm API key reset with token from email",
		Example: "  scraps key reset-confirm abc123token",
		Args: func(cmd *cobra.Command, args []string) error {
			if len(args) < 1 {
				return fmt.Errorf("reset token required\n\nUsage: scraps key reset-confirm <token>\n\nThe token is sent to your email after running 'scraps key reset-request'")
			}
			return nil
		},
		RunE: func(cmd *cobra.Command, args []string) error {
			token := args[0]

			if host == "" {
				host = config.GetHost()
			}

			client := api.NewClient(host, "")
			resp, err := client.ResetAPIKeyConfirm(token)
			if err != nil {
				return err
			}

			if !noLogin {
				// Save the new credentials
				if err := config.SetCredential(host, config.Credential{
					APIKey:   resp.APIKey,
					UserID:   resp.UserID,
					Username: resp.Username,
				}); err != nil {
					warn(fmt.Sprintf("Failed to save credentials: %v", err))
				} else {
					success(fmt.Sprintf("Logged in as %s", resp.Username))
				}
			}

			fmt.Printf("\nYour new API key: %s\n", resp.APIKey)
			fmt.Println("Save this key - it won't be shown again!")

			return nil
		},
	}

	cmd.Flags().StringVarP(&host, "host", "H", "", "Server host")
	cmd.Flags().BoolVar(&noLogin, "no-login", false, "Don't save credentials after reset")

	return cmd
}

File: src/commands/key.ts
import { Command } from "commander";
import { loadConfig, setCredential } from "../config.js";
import { ScrapsClient } from "../api.js";
import { success, error, info } from "../utils/output.js";

export function registerKeyCommands(program: Command): void {
  const key = program.command("key").description("API key management");

  key
    .command("reset-request <email>")
    .description("Request an API key reset email")
    .option("-h, --host <host>", "Server host")
    .addHelpText("after", `
Sends a reset link to your email address. Use the token from
the email with 'scraps key reset-confirm' to get a new API key.

Example:
  scraps key reset-request user@example.com
`)
    .action(async (email: string, opts) => {
      const config = loadConfig();
      const host = opts.host || config.default_host;
      const client = new ScrapsClient(host);

      try {
        await client.post("/api/v1/reset-api-key", { email });
        success(`Reset email sent to ${email}`);
        info(`Check your email and run: scraps key reset-confirm <token>`);
      } catch (e: any) {
        error(`Failed to request reset: ${e.message}`);
        process.exit(1);
      }
    });

  key
    .command("reset-confirm <token>")
    .description("Confirm API key reset with token from email")
    .option("-h, --host <host>", "Server host")
    .option("--no-login", "Don't save credential after reset")
    .addHelpText("after", `
Confirms your API key reset using the token from your email.
Your old API keys will be revoked and a new one will be returned.

Examples:
  scraps key reset-confirm abc123           # Reset and auto-login
  scraps key reset-confirm abc123 --no-login  # Just show the key
`)
    .action(async (token: string, opts) => {
      const config = loadConfig();
      const host = opts.host || config.default_host;
      const client = new ScrapsClient(host);

      try {
        const result = await client.get(
          `/api/v1/confirm-reset?token=${encodeURIComponent(token)}`
        );

        success("API key reset successful");
        info(`API key: ${result.api_key}`);

        if (opts.login !== false) {
          setCredential(host, {
            api_key: result.api_key,
            user_id: result.user_id,
            username: result.username,
          });
          info(`You are now logged in as ${result.username}`);
        }
      } catch (e: any) {
        error(`Failed to confirm reset: ${e.message}`);
        process.exit(1);
      }
    });
}
